pkg/operation/botanist/component/kubeapiserver/kube_apiserver.go

```diff
@@ -307,8 +309,10 @@ func (k *kubeAPIServer) Deploy(ctx context.Context) error {
 		return err
 	}
 
-	if err := k.reconcileSecretUserKubeconfig(ctx, secretStaticToken, secretBasicAuth); err != nil {
-		return err
+	if pointer.BoolDeref(k.values.EnableAdminKubeconfig, true) {
```

Comment:
I am missing a unit test case for this behaviour.

Reply:
Is [this](https://github.com/gardener/gardener/blob/18fa04ac2710ab344dddf6378ad36c87d9a5dea1/pkg/operation/botanist/component/kubeapiserver/kube_apiserver_test.go#L1958) not the correct unit test ?

Reply:
Right, sorry, I missed this case